base/src/main/java/io/spine/code/proto/enrichment/FieldRef.java

```diff
@@ -47,30 +51,35 @@
 
     /**
      * A delimiter between a field name and its qualifier, which refers to another message
-     * like {@code "context"}.
+     * like {@code "context"}, or between nested field names like {@code "timestamp.seconds"}.
      */
-    private static final String TYPE_SEPARATOR = ".";
+    private static final String SEPARATOR = ".";
 
     /**
-     * Separates a type name from a field name.
+     * Separates tokens in a field reference.
      */
-    private static final Splitter fieldNameSplit = Splitter.on(TYPE_SEPARATOR);
+    private static final Splitter split = Splitter.on(SEPARATOR).trimResults();
 
     /**
-     * Reference value parts separated by {@link #fieldNameSplit}.
+     * Reference to a containing type.
      */
-    private final ImmutableList<String> parts;
+    private final TypeRef typeRef;
 
     /**
-     * Reference to a containing type.
+     * {@code true} if the referenced field is nested inside another field(s value.
      */
-    private final TypeRef typeRef;
+    private final FieldPath path;
 
     @VisibleForTesting
     FieldRef(String value) {
         super(checkValue(value));
-        this.parts = split(value);
-        this.typeRef = TypeRef.parse(typeRef(value));
+        ImmutableList<String> parts = split(value);
+        this.typeRef = CONTEXT.parse(parts.get(0))
+                              .orElse(ANY);
+        // If the first element is context reference, skip it from the path.
+        ImmutableList<String> fieldPath =
+                parts.subList((this.typeRef == CONTEXT ? 1 : 0), parts.size());
```

Comment:
Why not use `isContext` here instead of equality check?

Reply:
Let's not escape the constructor here. There's no much sense.